### PR TITLE
Add Pylint CI

### PR DESCRIPTION
**Pull request by winstonsung**
Bug: #37
Bug: #38
Change-Id: I2e16cb11fc055c181df8234a8d3004977d352b02

## Summary by CodeRabbit

- **New Features**
  - Added a new GitHub Actions workflow to run Pylint for Python code analysis on various triggers.

- **Bug Fixes**
  - Updated the database context from "Discord" to "CTF" for managing user points.

- **Configuration**
  - Adjusted Pylint configuration settings for improved code linting, including formatting rules, naming styles, and linter checks.

## Sequence Diagram(s) (Beta)

```mermaid
sequenceDiagram
    actor Developer
    participant GitHub
    participant GitHub Actions
    participant Python Code
    Developer ->> GitHub: Push code / Create PR
    GitHub ->> GitHub Actions: Trigger Pylint workflow
    GitHub Actions ->> Python Code: Set up Python environment
    GitHub Actions ->> Python Code: Install dependencies
    GitHub Actions ->> Python Code: Run Pylint
    Python Code ->> GitHub Actions: Pylint results
    GitHub Actions ->> Developer: Report Pylint results
```

### CODE REVIEW

**Comment by Edit-Mr** [comment]
@winstonsung 應該 Test 3.11 就好了。Pycord 比較龜毛一點

**Comment by winstonsung** [comment]
> @winstonsung 應該 Test 3.11 就好了。Pycord 比較龜毛一點

Matrix changed to Python 3.11, 3.12.

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flask py-cord mysql-connector-python requests
+        pip install pylint
+    - name: Analysing the code with pylint
+      run: |
+        pylint $(git ls-files '*.py')
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply .github/workflows/pylint.yml:25]: Ignore.

File: .github/workflows/pylint.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flask py-cord mysql-connector-python requests
+        pip install pylint
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply .github/workflows/pylint.yml:22]: Done.

File: .github/workflows/pylint.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Pylint
+
+on:
+  [ pull_request, push, workflow_dispatch ]
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply .github/workflows/pylint.yml]: Done.

**Comment by winstonsung** [comment]
Finally passed!

https://github.com/winstonsung/scaict-uwu/actions/runs/9697254355

**Pull request merged**